components/content/src/utils.rs

```diff
@@ -58,6 +59,53 @@ pub fn find_related_assets(path: &Path, config: &Config, recursive: bool) -> Vec
     assets
 }
 
+/// Serializes assets source path for assets colocated with a section or a page
+pub fn serialize_assets(
+    assets: &Vec<PathBuf>,
+    parent_path: Option<&Path>,
+    colocated_path: Option<&String>,
+) -> Vec<String> {
+    assets
+        .iter()
+        .filter_map(|asset| asset.strip_prefix(parent_path.unwrap()).ok())
+        .map(|asset_relative_path| {
+            asset_relative_path
+                .components()
+                .map(|component| component.as_os_str().to_string_lossy().to_string())
+                .collect::<Vec<String>>()
+                .join("/")
+        })
+        .map(|asset_relative_path_as_string| {
+            format!(
+                "/{}{}",
+                colocated_path.expect("Should have colocated path for assets"),
```

Comment:
If it should have it, just pass &String for colocated_path instead of the Option

Reply:
If `assets` is empty, I don't have any guarantee that `colocated_path` will be Some.
I tried to reuse the same structure as the existing code but I can add an `if !assets.is_empty()` to unwrap this variable once instead of doing it in the iteration.
Doing it outside of `serialize_assets` to get rid of the `Option` in the signature would require to wrap the call of this function in the if statement mentionned above.
This would result in code duplication as I would have to do it twice, one for the pages and one for the sections.

What would you prefer me to implement?
1. Keep it as it is
2. Add the if statement in `serialize_assets` function to avoid unwrapping the option in the iteration
3. Add the if statement around the `serialize_assets`